Added horizontal separator and paragraph padding

ajolipa opened the pull request:
### In this PR
Adds a case for the type `horizontal-spacer` to the rich text renderer, and adds a bit of vertical padding around basic paragraph elements to better match the visual layout of the RTE in the admin client.

### Notes
This is related to Issue #127, which refers to specifically this page: https://tanyaclement.github.io/znh-1939/pages/introduction/ . Is the paragraph spacing something that we in fact want in general on all projects, or just something we want for that page? (Personally I like having the spacing between paragraphs, but just want to make sure since this is a general fix not just related to that project.)

camdendotlol commented:
The tradeoff to consider for vertical padding on paragraphs is that they will also appear in annotations and descriptions, where vertical space is limited. This is especially an issue when there are lots of newlines, e.g. poetry snippets will look really bad in the annotation list because each line is a `p` tag.

I'm fine with the margins if people think the tradeoff is worth it. Or we could use a global CSS rule to conditionally show the vertical padding depending on whether it's on a page or an annotation/description.

ajolipa commented:
> The tradeoff to consider for vertical padding on paragraphs is that they will also appear in annotations and descriptions, where vertical space is limited. This is especially an issue when there are lots of newlines, e.g. poetry snippets will look really bad in the annotation list because each line is a `p` tag.

Mmm yeah great point. Probably worth only doing conditionally for page bodies, which shouldn't be hard to do with global CSS as you said; I'll await @absempere's thoughts but that sounds reasonable to me.

lwjameson commented:
Agree with Camden here. Margins in pages makes sense, but. not elsewhere. We should strive to have custom pages appear as do in the editor of course.

absempere commented:
In general we want space between paragraphs, but how would that affect lines of poetry? Can we differentiate between what's essentially a `<br>` from a `<p> `with padding?

absempere commented:
Sorry, reading up, and I mean poetry snippets on pages rather than in annotations...

camdendotlol commented:
`<br>`s are supported in imported data (see https://github.com/AVAnnotate/admin-client/pull/194), but there's no way to add a `<br>` via the rich text editor.

We could probably add an "insert line break" button to the editor that inserts a new line without splitting the paragraph node. Or from a UX perspective it might be better if there were a toggle in the editor to switch between "pressing enter = new paragraph" and "pressing enter = just a `<br>`". Can't say offhand how complex that would be to implement though.

ajolipa commented:
Okay, I have updated it so that, as best I can determine, it will apply to descriptions and page content on custom or home pages, but not within annotations; will this work as a compromise for now, even though it will make poems inside page text have more spacing between lines? Or should we hold off on this entirely until we've implemented something like Camden outlined above? (Which does sound like a larger project...) @absempere Let me know!

absempere commented:
Let's go with the compromise version and revisit in future RTE work. Thanks, all!
